### PR TITLE
Fix slice-pairing algorithm to work regardless of line segment size.

### PR DESCRIPTION
**Pull request by maxvoxel8**

### CODE REVIEW

File: src/slice.jl [hunk 1]
```diff
@@ -68,39 +69,33 @@ function Base.slice(mesh::Mesh{Vector3{Float64}, Face{Int}}, heights::Vector{Flo
...
                     end
                 end
 
-                if (seg == start #We couldn't pair the segment
-                    || seg == lastseg) #The polygon can't be closed
+                if min_index == 0
                     break
                 end
+                push!(poly, lines[min_index])
+                paired[min_index] = true
```
Comment (dreammaker) [inline src/slice.jl:98]: `paired[min_index] = true` is set earlier now. This line seems redundant and/or contradictory with setting `paired[i] = true` below. What am I missing here?

**Comment by dreammaker** [comment]
This change is to prevent cases where `autoeps` was causing epsilon to become zero, correct?

It seems that the main change is that the `autoeps` parameter and calculation is removed.  Could we get the same effect by simply passing in `autoeps=false` when we call this function?

**Comment by sjkelly** [comment]
This is a non-trivial change for me. Can you describe the approach here a bit and cases the algorithm currently fails? I actually removed this code from GeometryTypes because I intend to rewrite the algorithm to be simpler for integers, so it would be helpful to have some edge cases.

**Comment by maxvoxel8** [comment]
The issue is that when autoeps is true, the eps value can in some cases shrink to less than the actual point tolerance, at which point polygons will not be successfully formed.

But then if autoeps is false, we need some other way to select the best segment to continue the polyline with. This should be the closest segment to the end of the line chain. Ideally we wouldn't need an eps at all, but we still want to have a way to distinguish between separate polygons.

I think an alternative approach might be to pass pair=false and write our own version of the algorithm in Euclid.

**Comment by sjkelly** [comment]
That makes sense. My current direction is to form all the segments into an
implicit function and let the GPU deal with it :p. However, based on the
issue you described, we should be able to make an approach that does not
track the pairing and simply prefers the closest point. I'll take a try at
this later today.

My only comment on this PR is that we should keep autoeps even though it
does nothing, just for API compatibility.

**Comment by dreammaker** [comment]
@sjkelly, after discussing this with @maxvoxel8, I think we're just going to write our own pairing.  So I'm closing this.

**Pull request closed without merging**